dev/com.ibm.ws.container.service.compat/src/com/ibm/ws/util/BoundedBuffer.java

```diff
@@ -91,8 +92,11 @@
      */
 
     // D312598 - begin
-    private static final int SPINS_TAKE_;
-    private static final int SPINS_PUT_;
+    private static AtomicInteger SPINS_TAKE_;
+    private static AtomicInteger SPINS_PUT_;
+    private static boolean spinsTakeProp = true;
```

Comment:
Ideally these booleans would be final.  Then you just change the code below to be:
```
spinsTakeProp = (spinsTake != null);
```

Reply:
The AtomicIntegers should be final as well.